cdap-master/src/main/java/co/cask/cdap/data/tools/Upgrade.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.data.tools;
+
+import com.google.inject.Injector;
+
+/**
+ * An interface to define Upgrade. Modules which needs to be upgraded should implement this.
+ */
+public interface Upgrade {
+  void upgrade(Injector injector) throws Exception;
```

Comment:
shouldn't take any args. Also, AbstractUpgrade should implement this.  Or you can just move the upgrade method to the abstract class and remove this interface.